src/commands/mount.rs

```diff
@@ -373,6 +371,37 @@ fn cmd_mount_inner(opt: Cli) -> Result<()> {
     }
 }
 
+fn attempt_unlock_master_key(passphrase_file: Option<PathBuf>, options: &String,  unlock_policy: UnlockPolicy, first_sb: bch_sb_handle) -> Result<KeyHandle, anyhow::Error> {
+    // Unlock by key_file CLI option
+    passphrase_file.and_then(|path|
+        match Passphrase::new_from_file(&first_sb, path) {
+            Ok(p) => Some(KeyHandle::new(&first_sb, &p)),
+            Err(e) => {
+                error!(
+                    "Failed to read passphrase from file, falling back to prompt: {}",
+                    e
+                );
+                None
+            }
+        }
+    )
+    // Unlock by key_file mount option
+    .unwrap_or_else(||parse_passphrase_file_from_mount_options(options).and_then(|path: PathBuf|
+        match Passphrase::new_from_file(&first_sb, path){
+            Ok(p) => Some(KeyHandle::new(&first_sb, &p)),
+            Err(e) => {
+                error!(
+                    "Failed to read passphrase from file, falling back to prompt: {}",
+                    e
+                );
+                None
+            }
+        }
+    )
+    // Unlock by unlock policy
+    .unwrap_or_else(|| unlock_policy.apply(&first_sb)))
```

Comment:
Not completely happy with this flow yet.

The opening bracket on 389 matches the last closing bracket at the end of this line (402).  Preferably I would close it on line 400, but haven't figured out how to get the types matched up quite yet.